packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -307,6 +306,7 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
   }, [isNotVisible, rowHeight, styleProp, minHeight, sizes, rootProps.rowSpacingType]);
 
   const rowClassNames = apiRef.current.unstable_applyPipeProcessors('rowClassName', [], rowId);
+  const ariaAttributes = rowNode ? getRowAriaAttributes(rowNode, index) : {};
```

Comment:
Nitpick: no need to create a new object here.
```suggestion
  const ariaAttributes = rowNode ? getRowAriaAttributes(rowNode, index) : undefined;
```